fb-negotiation/core/split_tracker.py

```diff
@@ -0,0 +1,327 @@
+import math
+import re
+import random
+from collections import defaultdict
+from itertools import chain
+from cocoa.core.util import read_json, write_pickle, read_pickle
+from tokenizer import tokenize
+
+class SplitTracker(object):
+  def __init__(self):
+    self.made = {'their_offer': False, 'their_offer_for_me': False}
+    self.their_offer = {'book':-1, 'hat':-1, 'ball':-1}
+    self.their_offer_for_me = {'book':-1, 'hat':-1, 'ball':-1}
+    self.items = ['book', 'hat', 'ball']
+    self.item_counts = None
+    self.needs_clarification = False
+    self.lexicon = []
+
```

Comment:
It seems this belongs to the rulebased system and maybe we should move it there? Because currently the tracker is optimized for the rulebased system and won't be used for other systems (e.g. IR or neural).

Reply:
Sure, I can move the split tracker over, but not too sure how you see this.  For example, do you just want the tracker code moved over while staying its own class, or merging the tracker functionality into System?  Maybe we can clarify this one more in person.